spec/services/media_storage/aws_adapter_spec.rb

```diff
@@ -18,20 +18,29 @@
   let(:adapter) { described_class.new(s3_opts) }
   let(:uri_regex) { /\A#{URI::DEFAULT_PARSER.make_regexp}\z/ }
 
-  context 'when keys are passed to the initializer' do
-    let(:expectation) do
-      expect(Aws::S3::Client)
-      .to receive(:new)
-      .with(s3_opts.except(:prefix, :bucket))
-      .and_call_original
-    end
+  context 'when opts are passed to the initializer' do
+    it "should use default expiration values" do
+      adapter = described_class.new(s3_opts)
+      default = MediaStorage::AwsAdapter::DEFAULT_EXPIRES_IN
+      expect(adapter.get_expiration).to eq(default)
+      expect(adapter.put_expiration).to eq(default)
+    end
+
+    describe "s3 client config" do
+      before do
+        expect(Aws::S3::Client)
+        .to receive(:new)
+        .with(s3_opts.except(:prefix, :bucket))
+        .and_call_original
+      end
 
-    it 'should set the aws config through the s3 client ' do
-      adapter
-    end
+      it 'should set the aws config through the s3 client ' do
+        adapter
```

Comment:
This style of tests continues to confuse me. I don't think I'll ever think to look for mock expectations in a before block. It makes the test examples confusing for me.

@zwolf  I need a tie-breaker on code style here.

Reply:
i didn't really like it either just saved some extra typing. i'll move it.